symengine/functions.cpp

```diff
@@ -2895,6 +2911,140 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
     return make_rcp<const UpperGamma>(s, x);
 }
 
+Beta::Beta(const RCP<const Basic> &x, const RCP<const Basic> &y)
+    : x_{x}, y_{y}
+{
+    SYMENGINE_ASSERT(is_canonical(x_, y_))
+}
+
+bool Beta::is_canonical(const RCP<const Basic> &x, const RCP<const Basic> &y)
+{
+    if (is_a<Integer>(*x) or (is_a<Rational>(*x) and
+        (rcp_static_cast<const Rational>(x)->i.get_den()) == 2)) 
+    {
+        if (is_a<Integer>(*y) or (is_a<Rational>(*y) and
+        (rcp_static_cast<const Rational>(y)->i.get_den()) == 2)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+std::size_t Beta::__hash__() const
+{
+    std::size_t seed = BETA;
+    hash_combine<Basic>(seed, *x_);
+    hash_combine<Basic>(seed, *y_);
```

Comment:
That's inefficient. Hashing is there for performance reasons. Adding/Xoring the hashes of the two arguments would be better.